infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/brokerphases/BrokerEnvironmentFactoryTest.java

```diff
@@ -238,4 +239,24 @@ public void shouldCreateConfigMapWithPluginFQNs() throws Exception {
               "Missing field from serialized config: expected '%s' in '%s'", expect, config));
     }
   }
+
+  @Test(dataProvider = "imageRefs")
+  public void testImageToContainerNameConversion(Object image, Object expected) {
+    String actual = factory.generateContainerNameFromImageRef((String) image);
+    assertEquals(
+        actual,
+        expected,
+        String.format("Should remove registry and organization from image '%s'.", image));
+  }
+
+  @DataProvider(name = "imageRefs")
+  public Object[][] imageRefs() {
+    return new Object[][] {
+      {"quay.io/eclipse/che-unified-plugin-broker:v0.20", "che-unified-plugin-broker-v0-20"},
+      {"very-long-registry-hostname-url.service/eclipse/image:tag", "image-tag"},
+      {"eclipse/che-unified-plugin-broker:v0.20", "che-unified-plugin-broker-v0-20"},
+      {"very-long-organization.name-eclipse-che/image:tag", "image-tag"},
+      {"very-long-registry-hostname-url.service/very-long-organization/image:tag", "image-tag"}
```

Comment:
different images from different registries or users will have same name

It looks odd when you debug if there is no easy way to know from where it could come

Reply:
You could always get the full image ref from the pod or env var though. There's no way to include registry and organization in the name, since it very easily goes over the 63 char limit (the length of just the broker pod name + version is 32 chars already).

Reply:
I never talked on including the full registry or org name
but I think it's like some logger are doing for packages
instead of seeing `org.apache.catalina.something` you see `o.a.c.something`

In logger of Tomcat I see for example `o.a.c.startup.VersionLoggerListener`
so I know more from where is coming the `VersionLoggerListener` class than if only `VersionLoggerListener` was displayed

So I would say you could do something similar, it's for UX.

and of course I could go to openshift webconsole or somewhere else but it's definitely not UX

Reply:
quay.io/my-organization/che-unified-plugin-broker:v0.20 --> `q-m-che-unified-plugin-broker-v0-20`

docker.io/my-organization/che-unified-plugin-broker:v0.20 --> `d-m-che-unified-plugin-broker-v0-20`

docker.io/other-organization/che-unified-plugin-broker:v0.20 --> `d-o-che-unified-plugin-broker-v0-20`

Reply:
@benoitf this sounds like a relevant improvement, but is it really that valuable?

Reply:
I think `q-m-che-unified-broker-image-v0-20` and `d-m-che-unified-plugin-broker-v0-20` are less readable than `che-unified-plugin-broker-v0-20`. Even with the truncation, it's an external reference (whereas logger trimming is referring to internal classes), so you'll still have to ask "what image is this/where is it hosted/what is it built from?" A single `m` character doesn't tell us anything real about the actual organization it's pushed to. I care more about what job the broker has (init vs unified), and what version it comes from (since e.g. `v0.19` brokers don't support offline).

Reply:
It seems i'm the only one to care UX or that it's just a hot fix of removing repo/org and not fixing the 63 limit

If we know that we can't reach more than 63 values should never return more than 63 characters


@amisevk. You may use other prefix it was an example to illustrate that you may care repo/org more than full image name as well

I won't block but to me it's not a fix of the issue. Just hot fix

Reply:
Also speaking about UX I think issue is that we're giving each image name independently and not a set of images so the logic can work only on a limited data while with all images it could dedup in a better format all images